lib/type-hinting.c

```diff
@@ -145,9 +145,9 @@ type_cast_to_double(const gchar *value, gdouble *out, GError **error)
 
   errno = 0;
   *out = strtod(value, &endptr);
-  if (errno == ERANGE && (*out == HUGE_VAL || *out == -HUGE_VAL))
+  if (errno == ERANGE && (*out >= HUGE_VAL || *out <= -HUGE_VAL))
     success = FALSE;
-  if (*out == 0 && endptr == value)
+  if ((int)(*out) == 0 && endptr == value)
```

Comment:
This confused me because I thought we would fail here with a float like 0.01, but the trick is endptr == value would tell us that any kind of parsing is happened. That led me thinking, why do we need to check *out == 0 at all? Would it be enough to check endptr == value? The code would be much cleaner.

Reply:
@furiel : You are correct. We can use endptr == value only to check if no conversion is performed.